File: ProblemSolving/DataStructures/Arrays/DynamicArray.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int a, n, q, x, y, idx;
    cin >> n >> q;

    vector<vector<int>>arr;
    for (int i=0; i<n; i++){
        vector<int>a;
        arr.push_back(a);
    }

    int lastanswer = 0;

    for(int i=0; i<q; i++){
        cin >> a >> x >> y;
        if (a == 1){
            idx = (x^lastanswer)%n;
            arr[idx].push_back(y);
        }
        else if (a == 2){
            idx = (x^lastanswer)%n;
            int l = y%arr[idx].size();
            lastanswer = arr[idx][l];
            cout << lastanswer << endl;
        }
    }

    return 0;
}

File: ProblemSolving/DataStructures/Arrays/DynamicArray.py
en, m = map(int, input().split())
arr = []

for i in range(n):
    a = []
    arr.append(a)

lastanswer = 0
for i in range(m):
    a, x, y = map(int, input().split())

    if a == 1:
        idx = (x^lastanswer)%n
        arr[idx].append(y)
    elif a == 2:
        idx =(x^lastanswer)%n
        l = y%len(arr[idx])
        lastanswer = arr[idx][l]
        #print("lastanswer:" ,lastanswer)
        print(lastanswer)
    #print(arr, sep = "\n")
        
